pkgs/applications/graphics/inkscape/extensions/textext.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, fetchFromGitHub
+, inkscape
+, texlive
+, python3
+}:
+
+let
+  texlive-combined = texlive.combined.scheme-basic;
+in
+python3.pkgs.buildPythonPackage rec {
+  pname = "textext";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "textext";
+    repo = "textext";
+    rev = "1.3.1";
+    sha256 = "0sq2gwlklgz55lqxrjnivjl8lmakr2c7dm3q10xlr0slhilrs7sv";
+  };
+
+  patches = [
+    # Make sure we can point directly to pdflatex in the extension,
+    # instead of relying on the PATH (which might not have it)
+    ./textext-placeholders.patch
+  ];
+
+  propagatedBuildInputs = [
+    # GTK is recommended, but I haven't looked into pkgs.pygobject3 yet
+    # https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-hooks-gobject-introspection
+    python3.pkgs.tkinter
+    texlive-combined
+  ];
+
+  # textext doesn't have a 'bdist_wheel' target
+  dontUseSetuptoolsBuild = true;
+
+  buildPhase = ''
+    runHook preBuild
+
+    mkdir dist
+
+    python setup.py \
+      --inkscape-executable=${inkscape}/bin/inkscape \
+      --pdflatex-executable=${texlive-combined}/bin/pdflatex \
+      --lualatex-executable=${texlive-combined}/bin/lualatex \
+      --inkscape-extensions-path=dist
+
+    runHook postBuild
+  '';
+
+  # textext doesn't have a 'test' target
+  doCheck = false;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/inkscape/extensions
+    cp -r dist/textext $out/share/inkscape/extensions
+
+    substituteInPlace $out/share/inkscape/extensions/textext/base.py \
+      --replace PDFLATEX_PLACEHOLDER ${texlive-combined}/bin/pdflatex \
+      --replace LUALATEX_PLACEHOLDER ${texlive-combined}/bin/lualatex
```

Comment:
Is it possible to do perform this replacement on the source in `postPatch` instead?
That would seem more robust, in case `pyc` files are produced, now or in a possible future.

Reply:
Even nicer would be using `substituteAll` on the patch itself.